aws-iot-scheduledaudit/src/main/java/com/amazonaws/iot/scheduledaudit/Translator.java

```diff
@@ -0,0 +1,109 @@
+package com.amazonaws.iot.scheduledaudit;
+
+import software.amazon.awssdk.services.iot.model.AuditFrequency;
+import software.amazon.awssdk.services.iot.model.DayOfWeek;
+import software.amazon.awssdk.services.iot.model.InternalFailureException;
+import software.amazon.awssdk.services.iot.model.InvalidRequestException;
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.awssdk.services.iot.model.LimitExceededException;
+import software.amazon.awssdk.services.iot.model.ResourceAlreadyExistsException;
+import software.amazon.awssdk.services.iot.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.iot.model.Tag;
+import software.amazon.awssdk.services.iot.model.ThrottlingException;
+import software.amazon.awssdk.services.iot.model.UnauthorizedException;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.exceptions.CfnAccessDeniedException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnInternalFailureException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceLimitExceededException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a centralized placeholder for
+ *  - api request construction
+ *  - object translation to/from aws sdk
+ *  - resource model construction for read/list handlers
+ */
+
+public class Translator {
```

Comment:
A general doubt since there are some classes like Translator, HandlerUtils which are same for all the resources but we have it defined separately for every resource, is it possible that we can have it in a common place?

Reply:
My thought on that is that it'd be complicated to set that up - each IoT resource has its own Maven project. So Translator and HandlerUtils would have to live in a separate project that all resources reuse. That would introduce complexity that's in my opinion not worth the amount of code that got duplicated.